File: Data Structures & Algorithms/search-insert-position/submission-2.cs
public class Solution {
    public int SearchInsert(int[] nums, int target) {
        int l = 0, r = nums.Length - 1,  m;

        while (l < r) {
            m = l + (r - l) / 2;

            if (nums[m] < target) {
                l = m + 1;
            }
            else {
                r = m;
            }
        }

        if (l == nums.Length - 1 && nums[l] < target) {
            l++;
        }

        return l;
    }
}

File: Data Structures & Algorithms/two-integer-sum/submission-2.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        indices = {}

        for i in range(len(nums)):
            if nums[i] not in indices:
                indices[nums[i]] = [i]
            else:
                indices[nums[i]].append(i)

        for i in range(0, len(nums)):
            remain = target - nums[i]
            if remain in indices:
                for j in indices[remain]:
                    if i != j:
                        return [i, j]
        
        return [-1, -1]
